src/main/java/org/magicdgs/thaplv/haplotypes/pairs/DifferencesDistancePair.java

```diff
@@ -0,0 +1,257 @@
+        }
+        // only compute the difference between called positions
+        // TODO: this is a putative broken change with respect to the previous repo
+        final int difference = AlleleUtils.scaledDifference(
+                haplotype1.getAllele(0), haplotype2.getAllele(0));
+        switch (difference) {
+            case 1: // missing
+                nMissing++;
+                break; // no site is count
+            case 2: // difference
+                differences++;
+            case 0: // no difference
+                nPositions++; // positions are count for both
+                break;
+            default:
+                // this should never happen
+                throw new RuntimeException("Unreacheable code");
+
+        }
+    }
+
+    /**
+     * Add a pair of haplotypes to the distance computation, checking for the names by default
+     *
+     * @param haplotype1 the first haplotype
+     * @param haplotype2 the second haplotype
+     *
+     * @throws IllegalArgumentException if the sample names does not match the ones in the pair
+     */
+    public void add(final Genotype haplotype1, final Genotype haplotype2) {
+        add(haplotype1, haplotype2, true);
+    }
+
+    /**
+     * Add a variant (which should contains both sample1 and sample2) to the distance computatio
+     *
+     * @param variant the variant which contains the haplotypes for the samples
+     *
+     * @throws java.lang.IllegalArgumentException if both sample names are not in the variant
+     */
+    public void add(final VariantContext variant) {
+        // false because it is already checked that this corresponds to this pair
+        add(variant.getGenotype(sample1), variant.getGenotype(sample2), false);
+    }
+
+    /**
+     * Add a difference of haplotypes to the distance computation, checking for the names by default
+     *
+     * @param haplotype  the sample haplotype
+     * @param checkNames {@code true} for check the names in the genotypes; {@code false }otherwise
+     *
+     * @throws IllegalArgumentException if the sample name does not match the ones in the pair
+     */
+    // TODO: probably this should be deprecated
+    // TODO: it is much better to create a fake haplotype for the reference
+    // TODO: and like that we will suport imput a different reference
+    protected void addReference(final Genotype haplotype, final boolean checkNames) {
+        // TODO: add check for this pair to assure that it is a reference one
+        if (checkNames && !containSample(haplotype.getSampleName())) {
+            logger.warn("Computing reference difference for ", haplotype.getSampleName(), " in ",
+                    getPairNames());
+            throw new IllegalArgumentException(
+                    "Trying to compute a difference between haplotypes that are not in this pair");
+        }
+        // only compute differences if it is called
+        // TODO: check this change wrt previous repo
```

Comment:
change for "// TODO: this is a putative broken change with respect to the previous repo"